### PR TITLE
[WIP] jsonEfication

### PR DESCRIPTION
**Pull request by owlishDeveloper**
https://bugzilla.mozilla.org/show_bug.cgi?id=1335915

I decided to start with a WIP PR as opposed to asking questions. This is how I currently imagine the place of the `json-e` library in life, Universe and everything. I hope that's correct? What puzzles me is passing tests 😄

I guess I also need to add some tests for the version 1 YAML.

### CODE REVIEW

File: src/intree.js [hunk 1]
```diff
@@ -103,14 +104,15 @@ module.exports.setup = function(cfg) {
...
-    }));
+    // config = jparam(config, _.merge(payload.details, {
+    //   $fromNow: (text) => tc.fromNowJSON(text),
+    //   timestamp: Math.floor(new Date()),
+    //   organization: payload.organization,
+    //   repository: payload.repository,
+    //   'taskcluster.docker.provisionerId': cfg.intree.provisionerId,
+    //   'taskcluster.docker.workerType': cfg.intree.workerType,
+    // }));
+    config = jsone(config);
```
Comment (djmitche) [inline src/intree.js]: jsone takes two inputs -- template and context.  What's in that context is the hard part.  You can pattern it after what we do in mozilla-taskcluster -- https://docs.taskcluster.net/reference/integrations/mozilla-taskcluster/docs/taskcluster-yml

Note that we will also need to render the template on the master branch, without any pushes, to look for `allowPullRequests` and other security-related configuration.  Maybe that can have a different `tasks_for`, perhaps `access_check`?

File: src/intree.js [hunk 2]
```diff
@@ -89,7 +90,7 @@ function completeInTreeConfig(config, payload) {
 module.exports.setup = function(cfg) {
   return function({config, payload, validator, schema}) {
     config = yaml.safeLoad(config);
-    let errors = validator(config, schema);
+    let errors = validator(config, schema); // TODO: add version 1 to the schema in base repo
```
Comment (djmitche) [inline src/intree.js]: ..and to the docs :)

File: src/intree.js (hunk 1 above)
Comment (owlishDeveloper) [reply src/intree.js]: Thank you Dustin! This is very helpful. I actually tried taking an initial stub at context, and got tangled :) I'll try hack on the context some time this week.

**Comment by owlishDeveloper** [comment]
Decided to push my work so far just to demonstrate I'm alive - there's nothing to review yet, I don't think.

**Comment by owlishDeveloper** [comment]
https://github.com/taskcluster/taskcluster-github/pull/247

**Pull request closed without merging**